src/ctelnet.cpp

```diff
@@ -2175,9 +2173,16 @@ void cTelnet::handle_socket_signal_readyRead()
 
     char in_bufferx[100010];
     char* in_buffer = in_bufferx;
-    char out_buffer[100010];
 
     int amount = socket.read(in_buffer, 100000);
+    readSocket(in_buffer, amount);
+}
+
+void cTelnet::readSocket(char*& in_buffer, int amount) {
```

Comment:
Cosmetic - but can we have the `{` on the next line please? :angel:

Less cosmetic but why have `in_buffer` at all now? As `in_bufferx` is a `char` array using `in_bufferx` as  `readSocket(in_bufferx, amount)` is the more normal way of passing C string pointers around - though the signature of `readSocket` should be amended to `(void) cTelnet::readSocket(char*, size_t)`...!  Of course since `socket.read(...)` can return a `qint64` value of `-1` on errors we really should ***not*** be calling the new method at all if `amount` is negative right there...

Reply:
Feedback applied!

Amount is checked for `-1` in the `readSocket()` a couple of lines below, so it's OK. Might as well have the check there because fuzzing could try that amount and we should be ready for it.